### PR TITLE
Group rework

### LINKED ISSUE #415: Group rework
We want to be replace groups in rubato with the more unity-esque functionality of gameobject parents and children.

This requires moving the current group functionality into scene, and then modifying gameobjects to work with the new grouping system.

This issue pairs well with the collision by tags issue #411

### PR DESCRIPTION
**Pull request by tsedan**
closes #415 

## Hello reviewers!
Before approving this PR, please ensure the following:

  - [ ] Documentation is up to date
  - [ ] Relevent tutorials are up to date
  - [ ] Changelog is up to date
  - [ ] That the new code actually works

Once all this is done, you can be sure that the PR is ready to be merged. 🎉

*This message has been added automatically. Please don't delete it.*
## Hello reviewers!
Before approving this PR, please ensure the following:

  - [ ] Documentation is up to date
  - [ ] Relevent tutorials are up to date
  - [ ] Changelog is up to date
  - [ ] That the new code actually works

Once all this is done, you can be sure that the PR is ready to be merged. 🎉

*This message has been added automatically. Please don't delete it.*

### CODE REVIEW

**Review by tinmarr — APPROVED** [review]
kachow

**Review by tinmarr — APPROVED** [review]

**Comment by tinmarr** [comment]
WAIT ON TOMER'S REVIEW BEFORE MERGING

File: CHANGELOG.md
Comment (mrElnekave) [inline CHANGELOG.md:41]: U sure about this? cannot say how helpful this was in debugging.

File: demo/platformer/end_menu.py
Comment (mrElnekave) [inline demo/platformer/end_menu.py:33]: Same as in discord: 
ok just an idea. Since this is a different behavious now (they will move w/ camera b/c we didn't ignore_cam them). We should still have an add_ui, that will just add normally and turn on the ignore_cam or we should turn on the ignore_cam for these in the demos

File: rubato/game.py
Comment (mrElnekave) [inline rubato/game.py:46]: Goog that it isn't in scene.

File: rubato/structure/gameobject/tilemap/tiled.py
Comment (mrElnekave) [inline rubato/structure/gameobject/tilemap/tiled.py:52]: still need to clone this in the clone, but I'll do that in reorder branch as I've done it already anyways.

Comment (tinmarr) [reply rubato/structure/gameobject/tilemap/tiled.py:52]: ok

File: CHANGELOG.md
Comment (tinmarr) [reply CHANGELOG.md:41]: if you need this you can set the name. Default name isn't useful

File: demo/platformer/end_menu.py
Comment (tinmarr) [reply demo/platformer/end_menu.py:33]: n o i dont want multiple functions

**Review by mrElnekave — DISMISSED** [review]
goog

**Review by tsedan — COMMENTED** [review]
gog

**Review by tinmarr — APPROVED** [review]

**Pull request merged**
